exchanges/bybit/bybit_wrapper.go

```diff
@@ -0,0 +1,1443 @@
+	}
+	if err != nil {
+		return book, err
+	}
+
+	for x := range orderbookNew.Bids {
+		book.Bids = append(book.Bids, orderbook.Item{
+			Amount: orderbookNew.Bids[x].Amount,
+			Price:  orderbookNew.Bids[x].Price,
+		})
+	}
+
+	for x := range orderbookNew.Asks {
+		book.Asks = append(book.Asks, orderbook.Item{
+			Amount: orderbookNew.Asks[x].Amount,
+			Price:  orderbookNew.Asks[x].Price,
+		})
+	}
+	err = book.Process()
+	if err != nil {
+		return book, err
+	}
+	return orderbook.Get(by.Name, formattedPair, assetType)
+}
+
+// UpdateAccountInfo retrieves balances for all enabled currencies
+func (by *Bybit) UpdateAccountInfo(ctx context.Context, assetType asset.Item) (account.Holdings, error) {
+	var info account.Holdings
+	var acc account.SubAccount
+	info.Exchange = by.Name
+	switch assetType {
+	case asset.Spot:
+		balances, err := by.GetWalletBalance(ctx)
+		if err != nil {
+			return info, err
+		}
+
+		currencyBalance := make([]account.Balance, len(balances))
+		for i := range balances {
+			currencyBalance[i] = account.Balance{
+				CurrencyName: currency.NewCode(balances[i].CoinName),
+				Total:        balances[i].Total,
+				Hold:         balances[i].Locked,
+				Free:         balances[i].Total - balances[i].Locked,
+			}
+		}
+
+		acc.Currencies = currencyBalance
+
+	case asset.CoinMarginedFutures, asset.USDTMarginedFutures, asset.Futures:
+		balances, err := by.GetFutureWalletBalance(ctx, "")
+		if err != nil {
+			return info, err
+		}
+
+		var i int
+		currencyBalance := make([]account.Balance, len(balances))
+		for coinName, data := range balances {
+			currencyBalance[i] = account.Balance{
+				CurrencyName: currency.NewCode(coinName),
+				Total:        data.WalletBalance,
+				Hold:         data.WalletBalance - data.AvailableBalance,
+				Free:         data.AvailableBalance,
+			}
+			i++
+		}
+
+		acc.Currencies = currencyBalance
+
+	default:
+		return info, fmt.Errorf("assetType not supported: %v", assetType)
+	}
+	acc.AssetType = assetType
+	info.Accounts = append(info.Accounts, acc)
+
+	if err := account.Process(&info); err != nil {
+		return account.Holdings{}, err
+	}
+	return info, nil
+}
+
+// FetchAccountInfo retrieves balances for all enabled currencies
+func (by *Bybit) FetchAccountInfo(ctx context.Context, assetType asset.Item) (account.Holdings, error) {
+	acc, err := account.GetHoldings(by.Name, assetType)
+	if err != nil {
+		return by.UpdateAccountInfo(ctx, assetType)
+	}
+
+	return acc, nil
+}
+
+// GetFundingHistory returns funding history, deposits and
+// withdrawals
+func (by *Bybit) GetFundingHistory(ctx context.Context) ([]exchange.FundHistory, error) {
+	return nil, common.ErrNotYetImplemented
+}
+
+// GetWithdrawalsHistory returns previous withdrawals data
+func (by *Bybit) GetWithdrawalsHistory(ctx context.Context, c currency.Code) (resp []exchange.WithdrawalHistory, err error) {
+	w, err := by.GetWalletWithdrawalRecords(ctx, "", "", "", "", 0, 0)
```

Comment:
I have understood your concern here. But I didn't understand what do you mean by "an expansion of the exchange interface properties might need to be implemented"? @gloriousCode

Reply:
Sorry for not being specific: https://github.com/thrasher-corp/gocryptotrader/blob/master/exchanges/interfaces.go#L66

For that line in question, we could add an asset.Item param for example: 
```diff
+ GetWithdrawalsHistory(ctx context.Context, code currency.Code, a asset.Item) ([]WithdrawalHistory, error)
- GetWithdrawalsHistory(ctx context.Context, code currency.Code) ([]WithdrawalHistory, error)
```

Reply:
Got it. Thanks for your explanation. 
Am I allowed to make necessary changes in interfaces.go? @gloriousCode 
I think it would also require updating the definition of `GetWithdrawalsHistory` across all exchanges.

Reply:
@Jdpurohit it's fine to make those changes and that's correct, updating it will require updating all exchanges

Reply:
Hello @gloriousCode 
I have completed all the review changes including the above.
I want your input on the above change request. This [commit](https://github.com/thrasher-corp/gocryptotrader/pull/887/commits/f0805fce17258f9f664f1ab9502ab3fbedba878c) includes the above change. 

Here I am unable to resolve the build issue in this file `engine/rpcserver.go`. Please provide your input.

Reply:
@Jdpurohit In `rpc.proto.go` please add `string asset_type = 5;` to [WithdrawalEventsByExchangeRequest](https://github.com/thrasher-corp/gocryptotrader/blob/master/gctrpc/rpc.proto#L534). In the `gtcrpc/` folder there is a generator `gen_pb_win.bat` or `gen_pb_linux.sh` script you can run that will update the underlying gRPC changes from the proto file. 

After you have done that change please head to `cmd/gctcli/` specifically [withdrawlRequestByExchangeID gctcli command](https://github.com/thrasher-corp/gocryptotrader/blob/master/cmd/gctcli/commands.go#L3087) where you will need to handle asset coming from the command line and add an additional paramater [here](https://github.com/thrasher-corp/gocryptotrader/blob/master/cmd/gctcli/commands.go#L3001) then test. 

Thank you sir!

Reply:
@shazbert Thank you once again for your explanation. 
I have made the change. Please review it and proceed further.